ls1intum/Artemis
# `Learning analytics`: Make competency taxonomies more user friendly

### Checklist
#### General


#### Client

### Motivation and Context



Selecting a taxonomy for a competency was not really user-friendly.

### Description


Artemis now suggests a potential taxonomy based on keywords used in the description.
The select input field makes the hierarchy of the taxonomies more clear.

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Navigate to Course Administration, Competencies, Create Competency
3. Select a taxonomy
4. Write a title or description with one or more keywords
5. Leave the input field to see that Artemis suggests a taxonomy

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage



Added a test case for the single newly introduced client side function.

### Screenshots



<img width="800" alt="Screenshot 2023-03-05 at 12 16 14" src="https://user-images.githubusercontent.com/102237363/222958473-1b48fb49-fd50-4dfa-8507-bd25742f3f3b.png">
